examples/csharp/Dockerfile

```diff
@@ -10,27 +11,21 @@ RUN dotnet tool install --global dotnet-sonarscanner \
 
 RUN dotnet tool list -g
 ENV PATH="${PATH}:/root/.dotnet/tools"
+ENV SONAR_TOKEN=$SONAR_TOKEN
 
 RUN mkdir /app
 WORKDIR /app
-RUN git clone --depth 1 https://github.com/twilio/twilio-csharp.git
+RUN git clone --depth 1 https://github.com/twilio/twilio-csharp.git -b csharp_monorepo_setup
 WORKDIR /app/twilio-csharp
 
 RUN rm -rf src/Twilio/Rest/*/
 COPY src/Twilio/Rest src/Twilio/Rest
 
 RUN rm -rf test/Twilio.Test/*/
-COPY unit-test/Rest integrationTest/Rest test/Twilio.Test/
+COPY unit-test/Rest test/Twilio.Test/
 
 RUN dotnet restore
 RUN dotnet build -c Release
-#RUN dotnet test test/Twilio.Test/Twilio.Test.csproj --no-build
-# sonar coverage reports
-# Add /d:sonar.login="${SONAR_TOKEN}" to sync up with sonarcloud.
-RUN dotnet sonarscanner begin /k:"twilio_twilio-csharp" /o:"twilio" /d:sonar.host.url=https://sonarcloud.io /d:sonar.language="cs" /d:sonar.exclusions="src/Twilio/Rest/**/*.*,test/Twilio.Test/**/*.*" /d:sonar.cs.opencover.reportsPaths="test/lcov.net451.opencover.xml"
-# /d:sonar.coverageReportPaths=".\sonarqubecoverage\SonarQube.xml"
-# Generates way too much logs / warnings so sending it to a log file instead
-RUN	dotnet build Twilio.sln > buildforsonar.log
-#RUN reportgenerator "-reports:./test/lcov.net451.opencover.xml" "-targetdir:test" "-reporttypes:SonarQube" -- enable this to generate better reports
-CMD ["/bin/bash", "-c", "dotnet test test/Twilio.Test/Twilio.Test.csproj --no-build /p:CollectCoverage=true /p:CoverletOutputFormat=opencover /p:CoverletOutput=../lcov| tee /local/test-report.out && cp test/lcov.net451.opencover.xml /local"]
-# dotnet sonarscanner end /d:sonar.login="${SONAR_TOKEN}" -- enable this when syncing up with sonarcloud
+
+RUN apt-get -y install build-essential
+RUN if [ "$SONAR_TOKEN" = "" ] ; then echo Skipping sonar analysis ;  else make cover PROJECT_NAME=twilio_twilio-oai-generator-csharp SONAR_SOURCES="/d:sonar.sources=src/Twilio/Rest /d:sonar.tests=test/Twilio.Test" ; fi
```

Comment:
Should still run the tests even if we're not reporting the coverage to sonar.

Reply:
Added command to run tests irrespective of value of sonar token